### DISCUSSION TITLE
Authentication changes in @supabase/ssr

### DISCUSSION THREAD

**Discussion by marcnajera** [supabase/supabase#19853] [Questions] [answered]
I recently migrated from @supabase/auth-helpers-sveltekit to @supabase/ssr and I haven't been able to register/authenticate users using server-side code.

For example: I used to use From Actions to perform user SignUps on the server (this allowed me to securely create a profile for them and redirect them afterwards in one go). That action would set a cookie and the user would be authenticated client-side after the redirection.

After a lot of headaches I realized that the cookies being set when using @supabase/ssr are HTTP-only, which causes that after submitting the SignUp form, creating the profile, and redirecting the user, they're not signed in (I guess because the SupabaseBrowserClient can't access those cookies to create a Session?).

I'm not an expert but I haven't found a way to use Form Actions as I used to, while signing in the user client-side.

Am I missing something? Is there a way to still do the Sign Up server-side and authenticate the user client-side at the same time? Or maybe a way to set those cookies non HTTP-only?

Thanks in advance!

PS: Among a ton of googling, I've read these two guides, but I haven't been able to find an answer:
- https://supabase.com/docs/guides/auth/server-side/migrating-to-ssr-from-auth-helpers?framework=sveltekit
- https://supabase.com/docs/guides/auth/server-side/creating-a-client?framework=sveltekit&environment=form-actions

PSS: For context, email confirmation is disabled: "enable_confirmations = false"

**Comment by silentworks** [comment, marked as answer]
By default the cookies are not HTTP-only in the latest version as this was a bug that got fixed. Please check you are using the latest version of the `@supabase/ssr` package and also remove your `node_modules` before re-installing it if your version number looks like the latest. You can see the cookie config constant here https://github.com/supabase/auth-helpers/blob/main/packages/ssr/src/utils/constants.ts#L6

**Reply by marcnajera** [reply]
Ohh! Thanks mate, having this functionality back is a game-changer 👍

**Comment by bombillazo** [comment]
Were you able to solve this? Since using the SSR package, our cookies are randomly erased from eh browser and the user is immediately logged out :/
